### PR TITLE
Parameterize queries by year

### PR DESCRIPTION
**Pull request by andrewparmet**
This change allows querying by year, e.g. by visiting `http://localhost:5000/?year=2019`.

It leaves the default as 2020 when no parameter is provided so it won't break existing behavior. It's a bit ugly to have `or 2020` all over the place - if the application script were modified to provide that query parameter, then the defaulting behavior can go away.

I had to do some renaming because of name clashes. Happy to change those names since I just hacked them in to stop causing issues.

This PR also adds a basic `.gitignore`.

Neat project!

### CODE REVIEW

File: imessage2020wrapped/sql_queries.py [hunk 1]
```diff
@@ -218,22 +257,28 @@
...
-    chat.chat_identifier as number
+    chat.chat_identifier as number,
+    date (date / 1000000000 + strftime ("%s", "2001-01-01"), "unixepoch", "localtime") as day
 FROM
     message
 JOIN chat_message_join cmj on cmj.message_id = message.ROWID
 JOIN chat on chat.ROWID = cmj.chat_id
 WHERE
-    chat_identifier REGEXP '\+1\d{10}'
+    chat_identifier REGEXP '\+1\d{{10}}'
```
Comment (michael-danello) [inline imessage2020wrapped/sql_queries.py:273]: Did you mean to change regex?

**Review by michael-danello — COMMENTED** [review]
Sorry that I'm getting to this so late...thanks for submitting the PR - it's a nice addition.

Largely looks LGTM - only question I had was about a non-year change in one of the queries

File: imessage2020wrapped/sql_queries.py (hunk 1 above)
Comment (andrewparmet) [reply imessage2020wrapped/sql_queries.py:273]: When you use `.format` you have to double-quote braces so the formatter doesn't confuse them with variable interpolations like I do below in `{year}` and `{next_year}`: https://stackoverflow.com/a/5466478.

Comment (michael-danello) [reply imessage2020wrapped/sql_queries.py:273]: Ah yep. Makes sense. Merging now.

**Pull request merged**